registry/remote/repository.go

```diff
@@ -399,13 +409,18 @@ func (r *Repository) Referrers(ctx context.Context, desc ocispec.Descriptor, art
 
 	// The referrers state is unknown.
 	if err != nil {
-		if errors.Is(err, errdef.ErrNotFound) {
-			// A 404 returned by Referrers API indicates that Referrers API is
-			// not supported. Fallback to referrers tag schema.
-			r.SetReferrersCapability(false)
-			return r.referrersByTagSchema(ctx, desc, artifactType, fn)
+		var errResp *errcode.ErrorResponse
+		if ok := errors.As(err, &errResp); !ok || errResp.StatusCode != http.StatusNotFound {
```

Comment:
According to the [doc](https://pkg.go.dev/errors#As), it is better to have
```suggestion
		if errors.As(err, &errResp) || errResp.StatusCode != http.StatusNotFound {
```